### PR TITLE
refs #000: unify check changelog job

### PR DESCRIPTION
**Pull request by Monska85**

### CODE REVIEW

File: templates/jobs/check_changelog.yml [hunk 1]
```diff
@@ -21,8 +22,10 @@
         echo "Checking presence of $CI_COMMIT_TAG in CHANGELOG.md"
         grep "${CI_COMMIT_TAG}" CHANGELOG.md || exit_code=$?
       else
-        echo "Checking if CHANGELOG.md has been modified since $CI_MERGE_REQUEST_DIFF_BASE_SHA"
-        git --no-pager diff --name-only "$(git merge-base "$CI_COMMIT_SHA" "$CI_MERGE_REQUEST_DIFF_BASE_SHA")" "$CI_COMMIT_SHA" | grep CHANGELOG.md || exit_code=$?
+        echo "Checking if CHANGELOG.md has been modified in the branch ${SRC_BRANCH}"
+        cd ${CI_PROJECT_DIR}
+        git fetch origin ${COMPARE_WITH}:refs/remotes/origin/${COMPARE_WITH}
+        git diff --name-only remotes/origin/${SRC_BRANCH}..remotes/origin/${COMPARE_WITH} | grep CHANGELOG.md >/dev/null || exit_code=$?
```
Comment (grota) [inline templates/jobs/check_changelog.yml:28]: I think this way of getting the diff is wrong logically, at least for the old scenario of the merge requests.

Imagine you start from master with your branch, then you take some time with your work (and not modify changelog in your branch). Meanwhile other contributions get merged into master that change changelog.
Getting the diff like this would encounter a change in changelog while in reality it should tell you there has been none.

That's why in the original job we take the "starting point of your work" with `$(git merge-base "$CI_COMMIT_SHA" "$CI_MERGE_REQUEST_DIFF_BASE_SHA")` 

Right now I don't recall why I used 
`git --no-pager diff --name-only "$(git merge-base "$CI_COMMIT_SHA" "$CI_MERGE_REQUEST_DIFF_BASE_SHA")" "$CI_COMMIT_SHA" `
instead of a simple
`git --no-pager diff --name-only $CI_MERGE_REQUEST_DIFF_BASE_SHA $CI_COMMIT_SHA`
I thought it was again to avoid false positives when you merge master in your long living feature branch, but the more I think about it the less it makes sense. It might be a overzealous copy-paste from stackoverflow. So I guess we can discard it and see what happens.

So to recap, I would suggest:
- adding `--no-pager`
- renaming the var SRC_BRANCH -> SRC_REF
- renaming COMPARE_WITH -> TO_REF
- using the syntax without the 2 dots
resulting in:

`git --no-pager diff --name-only "$SRC_REF" "$TO_REF"`
and in the jobs:

```
.check-changelog:
  variables:
    SRC_REF: $CI_MERGE_REQUEST_DIFF_BASE_SHA
    TO_REF: $CI_COMMIT_SHA
  extends:
    - .check-changelog-base
```

```
.check-changelog-standard-pipelines:
  variables:
    SRC_REF: remotes/origin/$CI_DEFAULT_BRANCH
    TO_REF: $CI_COMMIT_SHA
  extends:
    - .check-changelog-base
```

**Review by grota — CHANGES_REQUESTED** [review]

File: templates/jobs/check_changelog.yml (hunk 1 above)
Comment (grota) [reply templates/jobs/check_changelog.yml]: pushed a commit with the changes we discussed, let me know if it works :)
In case we can squash

**Review by grota — APPROVED** [review]

**Review by paolomainardi — APPROVED** [review]
